xilem_core/src/views/map_state.rs

```diff
@@ -42,28 +43,90 @@ pub struct MapState<ParentState, ChildState, V, F = fn(&mut ParentState) -> &mut
 pub fn map_state<ParentState, ChildState, Action, Context: ViewPathTracker, Message, V, F>(
     view: V,
     f: F,
-) -> MapState<ParentState, ChildState, V, F>
+) -> MapState<V, F, ParentState, ChildState, Action, Context, Message>
 where
     ParentState: 'static,
     ChildState: 'static,
     V: View<ChildState, Action, Context, Message>,
     F: Fn(&mut ParentState) -> &mut ChildState + 'static,
 {
     MapState {
-        f,
+        map_state: f,
         child: view,
         phantom: PhantomData,
     }
 }
 
-impl<ParentState, ChildState, V, F> ViewMarker for MapState<ParentState, ChildState, V, F> {}
-impl<ParentState, ChildState, Action, Context: ViewPathTracker, Message, V, F>
-    View<ParentState, Action, Context, Message> for MapState<ParentState, ChildState, V, F>
+/// An adapter which allows using a component which only uses one field of the current state.
+///
+/// In Xilem, many components are functions of the form `fn my_component(&mut SomeState) -> impl WidgetView<SomeState>`.
+/// For example, a date picker might be of the form `fn date_picker(&mut Date) -> impl WidgetView<Date>`.
+/// The `lens` View allows using these components in a higher-level component, where the higher level state has
+/// a field of the inner component's state type.
+/// For example, a flight finder app might have a `Date` field for the currently selected date.
+///
+/// The parameters of this view are:
+/// - `state`: The current outer view's state
+/// - `map`: A function from the higher-level state type to `component`'s state type
+/// - `component`: The child component the lens is being created for.
```

Comment:
Nice job on the docs!
(just a nit regarding ordering, so that this is in the same order as the function)
```suggestion
/// - `component`: The child component the lens is being created for.
/// - `state`: The current outer view's state
/// - `map`: A function from the higher-level state type to `component`'s state type
```